### ISSUE TITLE
Add ability to obtain available ports with magellan API

### ISSUE THREAD

**Issue by jhudson8** [TestArmada/magellan#112]
This would be useful if you have the need to start servers at the magellan level (rather than the individual test level which may be running in parallel).

My suggestion would be to add utility argument to the `initialize` function in the global setup/teardown (so it would remain backwards compatible).

For this to be of any value, there are at least 2 methods that should be available

* request allocation of a new open port
* apply additional CLI arg to test runner (nightwatch CLI argument for example)

Example usage is below

```
  var SetupTeardown = function () {
  };

  SetupTeardown.prototype = {

    initialize: function (util) {
        var appPort = util.allocatePort();
        var mocksPort = util.allocateMocksPort();
        // now start app and mock servers

        // but we also have to tell the nightwatch worker what the app port is (so the tests can refer to the appropriate URL
        util.addWorkerCLIArg('appPort', appPort);
    }
    ...
```

**Comment by dankle** [comment]
This would be very useful indeed.

**Comment by hjain60** [comment]
@jhudson8 : Would this mean I can start only one application and one mock server per worker and re-use them for multiple test cases?

**Comment by jhudson8** [comment]
It would mean you build a single set of application artifacts referring to a single mock server - you would no longer need to inject the mock port into the application artifacts per nightwatch worker.  You would still have an application server per worker but all apps would be talking to the same mock server (thus, allowing you to have a single application build and no worker-specific port injection)

**Comment by Maciek416** [comment]
This is implemented as of `8.5.0`. 

Note that `addWorkerCLIArg` has not been implemented -- The reason for this being that we can write to `process.env` to share information with child nodes, and this is a safe operation that doesn't require additional infrastructure in magellan plus update of plugins, etc. I hope that's okay.
